### PR TITLE
LVM-438-Refactored Exam to Assessment

### PR DESCRIPTION
**Pull request by xingyif**
Refactored Exam to Assessment - MVC+Migration:
https://cs5500-jira.ccs.neu.edu/browse/LVM-438

Changed exam_date column name to date:
https://cs5500-jira.ccs.neu.edu/browse/LVM-439

### CODE REVIEW

File: app/controllers/assessments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  end
+
+  private
+
+  def set_assessment
+    @assessment = Assessment.find(params[:id])
+  end
+
+  def assessment_params
+    params.require(:assessment).permit(:score, :assessment_date, :subject)
```
Comment (zack) [inline app/controllers/assessments_controller.rb]: `:assessment_date` -> `:date`?

File: app/views/assessments/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<h1>Editing assessment</h1>
+
+<%= render 'form', assessment: @assessment %>
+
+<%= link_to 'Show', @assessment %> |
+<%= link_to 'Back to assessments', assessments_path %>
```
Comment (zack) [inline app/views/assessments/edit.html.erb]: `Assessments`

File: db/fixtures/exams.rb [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
 20.times do
-  Exam.seed do |e|
-    e.subject = Faker::Internet.slug('English')
-    e.exam_date = Faker::Date.between(10.years.ago, Date.today)
-    e.score = Random.rand(100.0).truncate(2)
-    e.student_id = students.pop
-    e.tutor_id = tutors.pop
+  Assessment.seed do |a|
+    a.subject = Faker::Internet.slug('English')
+    a.assessment_date = Faker::Date.between(10.years.ago, Date.today)
```
Comment (zack) [inline db/fixtures/exams.rb]: `a.date`

File: db/fixtures/exams.rb [hunk 4]
```diff
@@ -1,14 +1,14 @@
-require 'exams_helper'
+require 'assessments_helper'
```
Comment (zack) [inline db/fixtures/exams.rb]: This file needs to be renamed

**Comment by zack** [comment]
Remaining locations of the word "Exam" in our app:

`app/views/coordinators/edit.html.erb:1` says "Editing Exam" (!?)
`app/views/coordinators/new.html.erb:1` says "New Exam" (!?)
`app/views/students/edit.html.erb:25` has a link to `/exams/new` and says "Exams" in the text. This should be a link helper and should not say "Exams"
`app/views/students/new.html.erb:23` has a link to `/exams/new` and says "Exams" in the text. This should be a link helper and should not say "Exams"
`db/schema.rb` shows that the students table has an exam_id column and index on it. These should be removed.
`db/schema.rb` shows that the tutors table has an exam_id column and index on it. These should be removed.

**Comment by zack** [comment]
nice job. almost there. :)

File: app/controllers/assessments_controller.rb (hunk 1 above)
Comment (xingyif) [reply app/controllers/assessments_controller.rb]: very very very good catch

**Review by zack — APPROVED** [review]
please remember to squash before merging!

**Pull request merged**